src-vue/src/components/ThunderstoreModCard.vue

```diff
@@ -0,0 +1,248 @@
+                    <Download />
+                </el-icon>
+                {{ modDownloadsCount }}
+            </span>
+
+            <span class="statContainer">
+                {{ mod.rating_score }}
+                <el-icon class="no-inherit">
+                    <Star />
+                </el-icon>
+            </span>
+            <br/>
+
+            <div class="name hide-text-overflow">{{ mod.name }}</div>
+            <div class="author hide-text-overflow">by {{ mod.owner }}</div>
+            <div class="desc">
+                {{ latestVersion.description }}
+            </div>
+
+            <span style="display: flex">
+                <el-button
+                    :type="modButtonType"
+                    style="flex: 6"
+                    :loading="isBeingInstalled || isBeingUpdated"
+                    @click.stop="installMod(mod)"
+                >
+                    {{ modButtonText }}
+                </el-button>
+                <el-button link type="info" class="infoBtn" @click="openURL(mod.package_url)">
+                    <el-icon>
+                        <InfoFilled />
+                    </el-icon>
+                </el-button>
+            </span>
+        </div>
+    </el-card>
+</template>
+
+<script lang="ts">
+import {defineComponent} from "vue";
+import {ThunderstoreMod} from "../utils/thunderstore/ThunderstoreMod";
+import {ThunderstoreModVersion} from "../utils/thunderstore/ThunderstoreModVersion";
+import {invoke, shell} from "@tauri-apps/api";
+import {ThunderstoreModStatus} from "../utils/thunderstore/ThunderstoreModStatus";
+import {NorthstarMod} from "../utils/NorthstarMod";
+import {GameInstall} from "../utils/GameInstall";
+import {ElNotification} from "element-plus";
+
+export default defineComponent({
+    name: "ThunderstoreModCard",
+    props: {
+        mod: {
+            required: true,
+            type: Object as () => ThunderstoreMod
+        }
+    },
+    data: () => ({
+        isBeingInstalled: false,
+        isBeingUpdated: false
+    }),
+    computed: {
+        latestVersion (): ThunderstoreModVersion {
+            return this.mod.versions[0];
+        },
+
+        /**
+         * Returns the status of a given mod.
+         */
+        modStatus(): ThunderstoreModStatus {
+            if (this.isBeingInstalled) {
+                return ThunderstoreModStatus.BEING_INSTALLED;
+            }
+            if (this.isBeingUpdated) {
+                return ThunderstoreModStatus.BEING_UPDATED;
+            }
+
+            // Ensure mod is up-to-date.
+            const tsModPrefix = this.getThunderstoreDependencyStringPrefix(this.latestVersion.full_name);
+            const matchingMods: NorthstarMod[] = this.$store.state.installed_mods.filter((mod: NorthstarMod) => {
+                if (!mod.thunderstore_mod_string) return false;
+                return this.getThunderstoreDependencyStringPrefix(mod.thunderstore_mod_string!) === tsModPrefix;
```

Comment:
`NorthstarMod`'s `thunderstore_mod_string` member is typed as such: `string | undefined`, meaning that it can be either a string or undefined.
That's why you can't invoke `method(thunderstore_mod_string)` directly; method takes a string as input, and you're not sure the input is an actual string.
By putting a `!` after your variable, you say that as a developer, you previously ensured that `thunderstore_mod_string` is not undefined, and it can be used as a string.